src/PVRIptvData.cpp

```diff
@@ -1482,3 +1487,26 @@ int PVRIptvData::GetChannelId(const char * strChannelName, const char * strStrea
 
   return abs(iId);
 }
+
+std::string& PVRIptvData::AddHeaderToStreamUrl(std::string& strLine, const std::string& header)
+{
+  size_t found = strLine.find('|');
+
+  if (found != std::string::npos)
+  {
+    size_t present = strLine.find(header + '=', found + 1);
+
+    if (present == std::string::npos)
+    {
+      strLine += '&';
+    }
+
+    return strLine;
```

Comment:
Why directly modify the reference and then return it?

Reply:
Because forgot putting the else.